### PR TITLE
[FMT] format example/02_decoding

### PR DESCRIPTION
**Pull request by Remi-Gau**
Relates to https://github.com/nilearn/nilearn/issues/2528

Changes proposed in this pull request:

- apply pre-commit hooks to example/02_decoding

### Files involved in other PRs:

- 'examples/02_decoding/plot_haxby_frem.py',
- 'examples/02_decoding/plot_haxby_searchlight_surface.py',

### CODE REVIEW

File: examples/02_decoding/plot_haxby_searchlight_surface.py [hunk 1]
```diff
@@ -9,66 +9,74 @@
...
 
-infl_mesh = fsaverage['infl_' + hemi]
+infl_mesh = fsaverage[f"infl_{hemi}"]
 coords, _ = surface.load_surf_mesh(infl_mesh)
-radius = 3.
+radius = 3.0
 nn = neighbors.NearestNeighbors(radius=radius)
 adjacency = nn.fit(coords).radius_neighbors_graph(coords).tolil()
 
+from sklearn.linear_model import RidgeClassifier
```
Comment (NicolasGensollen) [inline examples/02_decoding/plot_haxby_searchlight_surface.py:67]: I don't understand why this import statement was moved here... It should be below the comment (which will be parsed as the section title), right ?

File: examples/02_decoding/plot_miyawaki_encoding.py [hunk 2]
```diff
@@ -149,40 +161,54 @@
...
 
 # bring the scores into the shape of the background brain
 score_map_img = masker.inverse_transform(cut_score)
 
-thresholded_score_map_img = threshold_img(score_map_img, threshold=1e-6, copy=False)
+thresholded_score_map_img = threshold_img(
+    score_map_img, threshold=1e-6, copy=False
+)
+
+from nilearn.image import coord_transform
```
Comment (NicolasGensollen) [inline examples/02_decoding/plot_miyawaki_encoding.py:175]: Same here

File: examples/02_decoding/plot_miyawaki_reconstruction.py [hunk 3]
```diff
@@ -16,66 +16,73 @@
...
-encoding approach for the same dataset.
+For an encoding approach for the same dataset, see also
+:ref:`sphx_glr_auto_examples_02_decoding_plot_miyawaki_encoding.py`
 
 .. include:: ../../../examples/masker_note.rst
 
 """
 
+
+import sys
```
Comment (NicolasGensollen) [inline examples/02_decoding/plot_miyawaki_reconstruction.py:27]: Should be below comment. But I don't think this comment is useful at all, so we might be better removing the comment instead...

File: examples/02_decoding/plot_haxby_searchlight.py [hunk 4]
```diff
@@ -53,7 +55,7 @@
 # .astype() makes a copy.
 process_mask = get_data(mask_img).astype(int)
 picked_slice = 29
-process_mask[..., (picked_slice + 1):] = 0
+process_mask[..., (picked_slice + 1) :] = 0
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply examples/02_decoding/plot_haxby_searchlight.py:58]: Not sure but defo a black thing

```bash
black --diff tmp.py
--- tmp.py      2023-03-02 10:07:52.701237 +0000
+++ tmp.py      2023-03-02 10:07:54.130716 +0000
@@ -1 +1 @@
-process_mask[..., (picked_slice + 1):] = 0
+process_mask[..., (picked_slice + 1) :] = 0
would reformat tmp.py
```

File: examples/02_decoding/plot_miyawaki_reconstruction.py [hunk 5]
```diff
@@ -134,17 +141,18 @@ def flatten(list_of_2d_array):
...
+sys.stderr.write(f" Done ({time.time() - t0:.2f}s).\n")
 
 ############################################################################
 # We define our prediction function
 # -----------------------------------
 sys.stderr.write("Training classifiers... \r")
 t0 = time.time()
 
+from sklearn.feature_selection import SelectKBest, f_classif
+
```
Comment (NicolasGensollen) [inline examples/02_decoding/plot_miyawaki_reconstruction.py:153]: Same here, the comment only restates what the import statement is already saying.

File: examples/02_decoding/plot_miyawaki_reconstruction.py [hunk 6]
```diff
@@ -114,16 +125,18 @@ def flatten(list_of_2d_array):
 # 0 0 0 0 0 0 0 1 1 0
 # 0 0 0 0 0 0 0 0 1 1
 
-height_tf = (np.eye(y_cols) + np.eye(y_cols, k=1))[:y_cols - 1] * .5
+height_tf = (np.eye(y_cols) + np.eye(y_cols, k=1))[: y_cols - 1] * 0.5
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply examples/02_decoding/plot_miyawaki_reconstruction.py:122]: Same

```bash
black --diff --color tmp.py
--- tmp.py      2023-03-02 10:10:05.445856 +0000
+++ tmp.py      2023-03-02 10:10:19.336303 +0000
@@ -1,3 +1,3 @@
 
-height_tf = (np.eye(y_cols) + np.eye(y_cols, k=1))[:y_cols - 1] * .5
+height_tf = (np.eye(y_cols) + np.eye(y_cols, k=1))[: y_cols - 1] * 0.5
would reformat tmp.py
```

File: examples/02_decoding/plot_oasis_vbm.py [hunk 7]
```diff
@@ -43,15 +43,17 @@
 .. include:: ../../../examples/masker_note.rst
 
 """
+import matplotlib.pyplot as plt
```
Comment (NicolasGensollen) [inline examples/02_decoding/plot_oasis_vbm.py:46]: Same, I think we want this import below the comment

File: examples/02_decoding/plot_oasis_vbm.py [hunk 8]
```diff
@@ -168,29 +183,39 @@
...
 neg_log_pvals, t_scores_original_data, _ = permuted_ols(
-    age, data,  # + intercept as a covariate by default
+    age,
+    data,  # + intercept as a covariate by default
     n_perm=2000,  # 1,000 in the interest of time; 10000 would be better
-    verbose=1, # display progress bar
-    n_jobs=1)  # can be changed to use more CPUs
+    verbose=1,  # display progress bar
+    n_jobs=1,
+)  # can be changed to use more CPUs
```
Comment (NicolasGensollen) [inline examples/02_decoding/plot_oasis_vbm.py:193]: If we think this comment is really useful, then we should move it to the line just above now

File: examples/02_decoding/plot_oasis_vbm_space_net.py [hunk 9]
```diff
@@ -14,22 +14,27 @@
...
 dataset_files = datasets.fetch_oasis_vbm(
     n_subjects=n_subjects, legacy_format=False
 )
-age = dataset_files.ext_vars['age'].astype(float)
+age = dataset_files.ext_vars["age"].astype(float)
 age = np.array(age)
 gm_imgs = np.array(dataset_files.gray_matter_maps)
 
 
+from sklearn.model_selection import train_test_split
```
Comment (NicolasGensollen) [inline examples/02_decoding/plot_oasis_vbm_space_net.py:29]: should be below comment

**Comment by ymzayek** [comment]
Quick comment, before merging this I think we should make sure other PRs are not open on the example files. For example #3483 is working on a decoding example. We should try to minimize introducing conflicts in contributors PRs as was planned with our formatting strategy. (This goes for all the example formatting PRs of course though I haven't checked if there are other potential conflicts)

**Review by NicolasGensollen — COMMENTED** [review]
Thanks for working on this @Remi-Gau !
Just a few suggestions but LGTM otherwise !

**Comment by Remi-Gau** [comment]
> Quick comment, before merging this I think we should make sure other PRs are not open on the example files. For example #3483 is working on a decoding example. We should try to minimize introducing conflicts in contributors PRs as was planned with our formatting strategy. (This goes for all the example formatting PRs of course though I haven't checked if there are other potential conflicts)

So just to explain how those changes were done to try to minimize merge conflicts:

1. I set up the precommit config to not touch any files that were in any PR: see this comment (https://github.com/nilearn/nilearn/pull/3548/files#r1122405725)
2. Then used `pre-commit run -a` to autoformat as much as possible.
3. Applied flynt to convert to f-strings "en masse"
4. Fixed manually any remaining issues (flake8 mostly)
5. Applied small refactoring suggested by sourcery
6. Only committed the files for a given example folder (had tried to them all in on PR but it would have hell to review)
7. The PR [for example/07_advanced](https://github.com/nilearn/nilearn/pull/3556) is a WIP because it also contains the CI changes to check all the examples folders: this PR will have to merged last.

**Comment by ymzayek** [comment]
@Remi-Gau thanks for the detailed explanation! All sounds good to me in that case :)

**Comment by Remi-Gau** [comment]
> @Remi-Gau thanks for the detailed explanation! All sounds good to me in that case :)

I think that it will be hard to get 0% merge conflict through the reformatting process.

Those examples PR may have a couple of changes on files involved somewhere else.

Will try to flag those whenever I can

**Comment by Remi-Gau** [comment]
> @Remi-Gau thanks for the detailed explanation! All sounds good to me in that case :)

@ymzayek I upgraded the script I used to list the files in PR to also tell us the files that are in several PR.

For each formatting PR I have added a list of the files that are involved in at least one other PR: this way we can try to minimize how much change is done on those files.

File: examples/02_decoding/plot_haxby_full_analysis.py [hunk 10]
```diff
@@ -58,78 +65,99 @@
...
-                             groups=session_labels)
-        mask_chance_scores[mask_name][category] = dummy_classifier.cv_scores_[1]
+        print(
+            "Scores: %1.2f +- %1.2f"
+            % (
+                np.mean(mask_scores[mask_name][category]),
+                np.std(mask_scores[mask_name][category]),
+            )
+        )
+
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply examples/02_decoding/plot_haxby_full_analysis.py]: should be fixed

File: examples/02_decoding/plot_haxby_different_estimators.py [hunk 11]
```diff
@@ -58,74 +65,92 @@
...
-            category,
-            np.mean(classifiers_data[classifier_name]['score'][category]),
-            np.std(classifiers_data[classifier_name]['score'][category]))
+        print(
+            "    %14s vs all -- AUC: %1.2f +- %1.2f"
+            % (
+                category,
+                np.mean(classifiers_data[classifier_name]["score"][category]),
+                np.std(classifiers_data[classifier_name]["score"][category]),
+            )
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply examples/02_decoding/plot_haxby_different_estimators.py]: should be fixed

File: examples/02_decoding/plot_miyawaki_reconstruction.py [hunk 12]
```diff
@@ -154,14 +168,20 @@ def flatten(list_of_2d_array):
...
+    )
     clf.fit(X_train, y_train[:, i])
     clfs.append(clf)
 
-sys.stderr.write("Training classifiers %03d/%d... Done (%.2fs).\n" % (
-    n_clfs, n_clfs, time.time() - t0))
+sys.stderr.write(
+    "Training classifiers %03d/%d... Done (%.2fs).\n"
+    % (n_clfs, n_clfs, time.time() - t0)
+)
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply examples/02_decoding/plot_miyawaki_reconstruction.py]: should be fixed

**Review by bthirion — APPROVED** [review]
LGTM

File: examples/02_decoding/plot_haxby_searchlight_surface.py (hunk 1 above)
Comment (Remi-Gau) [reply examples/02_decoding/plot_haxby_searchlight_surface.py]: I have tried to move the import in sensible place and re ran isort to make sure it was happy with it.

File: examples/02_decoding/plot_miyawaki_encoding.py (hunk 2 above)
Comment (Remi-Gau) [reply examples/02_decoding/plot_miyawaki_encoding.py]: I have tried to move the import in sensible place and re ran isort to make sure it was happy with it.

File: examples/02_decoding/plot_miyawaki_reconstruction.py (hunk 3 above)
Comment (Remi-Gau) [reply examples/02_decoding/plot_miyawaki_reconstruction.py:25]: did that

File: examples/02_decoding/plot_oasis_vbm.py (hunk 7 above)
Comment (Remi-Gau) [reply examples/02_decoding/plot_oasis_vbm.py]: done

File: examples/02_decoding/plot_miyawaki_reconstruction.py (hunk 5 above)
Comment (Remi-Gau) [reply examples/02_decoding/plot_miyawaki_reconstruction.py]: done

File: examples/02_decoding/plot_oasis_vbm_space_net.py (hunk 9 above)
Comment (Remi-Gau) [reply examples/02_decoding/plot_oasis_vbm_space_net.py]: done

**Review by ymzayek — APPROVED** [review]
Ok to merge once the CI passes. Thx @Remi-Gau

**Pull request merged**